src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -489,8 +490,8 @@ bool CartGridStrategy::doProbe(Gcode *gc)
     setAdjustFunction(false);
     reset_bed_level();
 
-    if(gc->has_letter('I')) current_grid_x_size = gc->get_value('I'); // override default grid x size
-    if(gc->has_letter('J')) current_grid_y_size = gc->get_value('J'); // override default grid y size
+    current_grid_x_size = gc->has_letter('I') ? gc->get_value('I') : (gc->has_letter('J') ? gc->get_value('J') : current_grid_x_size); // If I provided update x grid size, if not try J.
+    current_grid_y_size = gc->has_letter('J') ? gc->get_value('J') : (gc->has_letter('I') ? gc->get_value('I') : current_grid_y_size); // If J provided update y grid size, if not try I. 
```

Comment:
I do not really agree with this change. I and J must be optional but if one is left out the other should revert to the default in config not to the one specified.
@lordofhyphens is this really what you wanted?